src/pages/playground/Playground.tsx

```diff
@@ -362,15 +396,11 @@ const Playground: React.FC<PlaygroundProps> = props => {
         // This is because Folder mode only works in Source 2+.
         dispatch(setFolderMode(workspaceLocation, false));
       }
-      return;
     }
-    handleHash(
-      hash,
-      { handleChangeExecTime, handleChapterSelect },
-      workspaceLocation,
-      dispatch,
-      fileSystem
-    );
+    handleURL();
+    return;
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
There's a reason the lint warning exists. Please fix the warning instead of disabling it.